include/test/battle.h

```diff
@@ -839,6 +841,8 @@ struct moveWithPP {
 #define Status1(status1) Status1_(__LINE__, status1)
 #define OTName(otName) do {static const u8 otName_[] = _(otName); OTName_(__LINE__, otName_);} while (0)
 
+#define FLAG_SET(flagId) do { FlagSet(flagId); gBattleTestRunnerState->data.setFlag = flagId; } while (0)
```

Comment:
I think I'd move the implementation into `test/test_runner_battle.h`, where you could take advantage of `INVALID_IF` to report an error if `FLAG_SET` is called twice in the same test (i.e. called with `gBattleTestRunnerState->data.setFlag != 0`).

Reply:
Also, could you document `FLAG_SET` at the top of this file, including how it works with `PARAMETRIZE`/`PASSES_RANDOMLY`.

I'm not sure that how it works with `PARAMETRIZE` is particularly intuitive—I think the flag stays set for all later `PARAMETRIZE`s, unless they happen to use `PASSES_RANDOMLY` in which case the flag is reset after every trial?